Source/Diagnostics/Diagnostics.cpp

```diff
@@ -0,0 +1,74 @@
+
+#include "Diagnostics.H"
+#include "WarpX.H"
+#include "Average.H"
+
+using namespace amrex;
+
+Diagnostics::Diagnostics (int i, std::string name)
+    : diag_index(i), diag_name(name)
+{}
+
+Diagnostics::~Diagnostics()
+{
+    delete m_flush_format;
+    /*
+    for ( int lev=0; lev<nlev; lev++ ){
+        for ( int i=0; i<ncomp; i++ ){
+            allfields[lev][i].reset();
+        }
+    }
+    */
+}
+
+void
+Diagnostics::InitData ()
+{
+    Print()<<"Diagnostics::InitData\n";
```

Comment:
Did you mean to leave this in?

Reply:
This should be called only if the user adds things like 
```
diagnostics.ndiags = 1
diagnostics.diags_names = d1
d1.diag_type = Full
```
which are undocumented parameters, so since this piece of code is still largely under development I thought we could keep this here... But I can also remove it and re-add when working on the next steps, anything's fine for me.

Reply:
Just checking - it's fine with me if you want to leave it in.